src/mlpack/methods/reinforcement_learning/training_config.hpp

```diff
@@ -0,0 +1,124 @@
+/**
+ * @file training_config.hpp
+ * @author Shangtong Zhang
+ *
+ * This file is the implementation of TrainingConfig class,
+ * which contains hyper-parameters for training RL agent.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_RL_TRAINING_CONFIG_HPP
+#define MLPACK_METHODS_RL_TRAINING_CONFIG_HPP
+
+namespace mlpack {
+namespace rl {
+
+class TrainingConfig
```

Comment:
I should have mentioned that, I was referring to the standard mlpack interface: especially "Hyperparameters are passed to the constructor, but not to Train().", if we do that we should probably do this for all methods to avoid any confusion. But I I think we can do this later and I agree we don't have to wait for the change.